engineio/transport/polling/encoder.go

```diff
@@ -0,0 +1,132 @@
+package polling
+
+import (
+	"encoding/base64"
+	"io"
+	"sync/atomic"
+	"time"
+
+	"github.com/googollee/go-socket.io/engineio/frame"
+)
+
+// encoder encodes frames to the writer.
+type encoder struct {
+	pingTimeout       time.Duration
+	lastPing          time.Time
+	writer            *bufWriter
+	hasFramesChan     chan struct{}
+	closed            chan struct{}
+	hasNonClosedFrame int32
+}
+
+func newEncoder(pingTimouet time.Duration, closed chan struct{}, buf []byte) *encoder {
```

Comment:
why are you need trow `closed` channel and `pingTimeout` in function interface?

i think `pingTimeout` is optional value

Reply:
No. Encoder handles timeout in `WriteFramesTo()`. The transport will block on `WriteFramesTo()` to write something out. If it doesn't have data to write, then returns timeout error to process timeout.

Reply:
i suggest change signature to:
```go
func newEncoder(buf []byte,  closed chan struct{}, pingTimeout time.Duration) *encoder {
```
i mean that use config/options type into encode struct with default `pintTimeout`, but now we use one optional value. at now, it is doesn't necessary